earth2studio/models/px/fcn3.py

```diff
@@ -0,0 +1,357 @@
+# SPDX-FileCopyrightText: Copyright (c) 2024-2025 NVIDIA CORPORATION & AFFILIATES.
+# SPDX-FileCopyrightText: All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import json
+from collections import OrderedDict
+from collections.abc import Generator, Iterator
+from datetime import datetime
+from importlib.utils import find_spec
+
+import numpy as np
+import torch
+from loguru import logger
+
+from earth2studio.models.auto import AutoModelMixin, Package
+from earth2studio.models.batch import batch_coords, batch_func
+from earth2studio.models.px.base import PrognosticModel
+from earth2studio.models.px.utils import PrognosticMixin
+from earth2studio.utils import handshake_coords, handshake_dim
+from earth2studio.utils.imports import (
+    OptionalDependencyFailure,
+    check_optional_dependencies,
+)
+from earth2studio.utils.time import timearray_to_datetime
+from earth2studio.utils.type import CoordSystem
+
+try:
+    from makani.models.model_package import load_model_package
+except ImportError:
+    OptionalDependencyFailure("fcn3")
+    load_model_package = None
+
+VARIABLES = [
+    "u10m",
+    "v10m",
+    "u100m",
+    "v100m",
+    "t2m",
+    "msl",
+    "tcwv",
+    "u50",
+    "u100",
+    "u150",
+    "u200",
+    "u250",
+    "u300",
+    "u400",
+    "u500",
+    "u600",
+    "u700",
+    "u850",
+    "u925",
+    "u1000",
+    "v50",
+    "v100",
+    "v150",
+    "v200",
+    "v250",
+    "v300",
+    "v400",
+    "v500",
+    "v600",
+    "v700",
+    "v850",
+    "v925",
+    "v1000",
+    "z50",
+    "z100",
+    "z150",
+    "z200",
+    "z250",
+    "z300",
+    "z400",
+    "z500",
+    "z600",
+    "z700",
+    "z850",
+    "z925",
+    "z1000",
+    "t50",
+    "t100",
+    "t150",
+    "t200",
+    "t250",
+    "t300",
+    "t400",
+    "t500",
+    "t600",
+    "t700",
+    "t850",
+    "t925",
+    "t1000",
+    "q50",
+    "q100",
+    "q150",
+    "q200",
+    "q250",
+    "q300",
+    "q400",
+    "q500",
+    "q600",
+    "q700",
+    "q850",
+    "q925",
+    "q1000",
+]
+
+
+@check_optional_dependencies()
+class FCN3(torch.nn.Module, AutoModelMixin, PrognosticMixin):
+    """
+
+    FourCastNet 3 global probabilisticprognostic model.
+    Consists of a single model with a time-step size of 6 hours.
+    FourCastNet 3 operates on 0.25 degree lat-lon grid (south-pole excluding)
+    equirectangular grid with 72 variables.
```

Comment:
Can you add a note to the API docs here that link to the paper / NGC page like:
https://github.com/NVIDIA/earth2studio/blob/main/earth2studio/models/px/fcn.py#L78

If the NGC page doesnt mention anything about the needed card size, May also want to add a note / warning that this requires a larger GPU card.

Reply:
Done.